apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarWalletSelector.tsx

```diff
@@ -0,0 +1,118 @@
+import { SidebarWalletSelectorFragment$key } from '~/generated/SidebarWalletSelectorFragment.graphql';
+import DoubleArrowsIcon from '~/icons/DoubleArrowsIcon';
+import { useTrack } from '~/shared/contexts/AnalyticsContext';
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+
+export type SidebarWallet = { chainAddress: { chain: string; address: string } } | 'All';
+
+type SidebarWalletSelectorProps = {
+  queryRef: SidebarWalletSelectorFragment$key;
+  selectedChain: ChainMetadata;
+  selectedWallet: SidebarWallet;
+  onSelectedWalletChange: (selectedWallet: SidebarWallet) => void;
+};
+
+export default function SidebarWalletSelector({
+  queryRef,
+  selectedChain,
+  selectedWallet,
+  onSelectedWalletChange,
+}: SidebarWalletSelectorProps) {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const { viewer } = useFragment(
+    graphql`
+      fragment SidebarWalletSelectorFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              wallets {
+                chainAddress @required(action: THROW) {
+                  address @required(action: THROW)
+                  chain @required(action: THROW)
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+  const userWalletsOnSelectedNetwork = useMemo(
+    () =>
+      removeNullValues(viewer?.user?.wallets).filter((wallet) => {
+        if (wallet.chainAddress.chain === chainsMap[selectedChain.name].baseChain) {
+          return wallet;
+        }
+      }),
+    [viewer?.user?.wallets, selectedChain]
+  );
+
+  const track = useTrack();
+
+  const onSelectWallet = useCallback(
+    (selectedWallet: SidebarWallet) => {
+      track('Editor Sidebar Wallet Dropdown Clicked', { variant: selectedWallet });
+      onSelectedWalletChange(selectedWallet);
+      setIsDropdownOpen(false);
+    },
+    [track, onSelectedWalletChange]
+  );
+
+  const displayAddressVal = (wallet: SidebarWallet) => {
+    if (wallet && wallet !== 'All') {
+      const address = wallet?.chainAddress.address;
+      return `${address.slice(0, 6)}...${address.slice(-4)}`;
+    }
+    return 'All';
+  };
+
+  return (
+    <Container>
+      <Selector gap={10} align="center" onClick={() => setIsDropdownOpen(true)}>
+        <BaseM>{displayAddressVal(selectedWallet)}</BaseM>
+        <IconContainer variant="stacked" size="sm" icon={<DoubleArrowsIcon />} />
+      </Selector>
+      <StyledDropdown
+        position="right"
+        active={isDropdownOpen}
+        onClose={() => setIsDropdownOpen(false)}
+      >
+        <DropdownSection>
+          <DropdownItem onClick={() => onSelectWallet('All')}>
+            <BaseM>All</BaseM>
+          </DropdownItem>
+          {userWalletsOnSelectedNetwork.map((wallet, index) => (
+            <DropdownItem key={index} onClick={() => onSelectWallet(wallet)}>
+              <BaseM>{displayAddressVal(wallet)}</BaseM>
```

Comment:
generally an antipattern to call functions directly in the render; this should `useMemo` instead